File: src/main/java/loxal/epvin/datastore/service/AppUserDAO.java
/*
 * Copyright 2012 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package loxal.epvin.datastore.service;

import com.googlecode.objectify.Objectify;
import com.googlecode.objectify.ObjectifyService;
import loxal.epvin.datastore.model.AppUser;

import java.util.List;

/**
 * @author Alexander Orlov <alexander.orlov@loxal.net>
 */
public class AppUserDAO extends DAO {
    final Objectify ofy = ObjectifyService.ofy();

    public void put(final AppUser appUser) {
        ofy.save().entity(appUser).now();
    }

    public AppUser get(final Long id) {
        return ofy.load().type(AppUser.class).id(id).now();
    }

    public List<AppUser> retrieve() {
        return ofy.load().type(AppUser.class).list();
    }
}

File: src/main/kotlin/loxal/epvin/datastore/service/AppUserDAO.kt
/*
 * Copyright 2015 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package loxal.epvin.datastore.service

import com.googlecode.objectify.ObjectifyService
import loxal.epvin.datastore.model.AppUser


public class AppUserDAO : DAO() {
    val ofy = ObjectifyService.ofy()

    public fun put(appUser: AppUser) {
        ofy.save().entity<AppUser>(appUser).now()
    }

    operator public fun get(id: Long?): AppUser {
        return ofy.load().type<AppUser>(AppUser::class.java).id(id!!).now()
    }

    public fun retrieve(): List<AppUser> {
        return ofy.load().type<AppUser>(AppUser::class.java).list()
    }
}